### PR TITLE
noDockOnStop -> DockOnStop

### PR DESCRIPTION
**Pull request by rcoletti116**
Changes:
config.schema.json
accessory.ts
ReadMe

Update Config option to DockOnStop, where 'true' means the robot will return home to it's dock when the HomeKit accessory is powered off/stopped.

The default value is set to 'true' so that the current expected behavior does not change for all users.

### CODE REVIEW

File: config.schema.json [hunk 1]
```diff
@@ -55,9 +55,10 @@
                 "title": "Show docking status as a contact sensor",
                 "required": false
             },
-            "noDockOnStop": {
+            "DockOnStop": {
```
Comment (karlvr) [inline config.schema.json:58]: I'd like to follow the same style as the other config properties and start with a lowercase d

File: src/accessory.ts [hunk 2]
```diff
@@ -431,8 +431,13 @@ export default class RoombaAccessory implements AccessoryPlugin {
 
                     if (state.running) {
                         this.log("Roomba is pausing");
-
                         await roomba.pause();
+                        if (this.DockOnStop) {
```
Comment (karlvr) [inline src/accessory.ts:435]: I didn't think we should wait for the pause to succeed before calling back to Homebridge. I'm not sure though. Perhaps we should merge the state and then callback, and _then_ actually send the command. 

Or maybe it's okay for setter callbacks to be called after some time. 

What are your thoughts around making this change?

**Review by karlvr — CHANGES_REQUESTED** [review]

File: src/accessory.ts (hunk 2 above)
Comment (rcoletti116) [reply src/accessory.ts]: I'm fine to revert the order if you think that's best. The main reason for making this change was due to the responsiveness of the robot, the home app, and their synchronization. However, that could have more to do with the connections than anything. The differences are minimal.

As far as a best practice related to HomeKit and the cached status - that I'm not sure.

File: src/accessory.ts [hunk 3]
```diff
@@ -128,7 +128,7 @@ export default class RoombaAccessory implements AccessoryPlugin {
         this.robotpwd = config.robotpwd;
         this.ipaddress = config.ipaddress;
         this.firmware = "N/A";
-        this.noDockOnStop = config.noDockOnStop;
+        this.dockOnStop = config.dockOnStop;
```
Comment (karlvr) [inline src/accessory.ts:131]: Does this work? I thought we needed to manually introduce the default of it being `true` like:

```typescript
this.dockOnStop = config.dockOnStop !== undefined ? config.dockOnStop : true;
```

But does Homebridge enforce the schema?

File: README.md [hunk 4]
```diff
@@ -84,7 +84,7 @@ configure your accessory using JSON:
 |`runningContactSensor`|Add a contact sensor to HomeKit that's _open_ when Roomba is running|
 |`binContactSensor`|Add a contact sensor to HomeKit that's _open_ when Roomba's bin is full|
 |`dockingContactSensor`|Add a contact sensor to HomeKit that's _open_ when Roomba is docking|
-|`noDockOnStop`|Do not send home when stopped|
+|`dockOnStop`|Sends roomba home to dock when stopped|
```
Comment (karlvr) [inline README.md:87]: I think this suggests we add another column with the default value, e.g.

```
|Key|Description|Default value|
...
|`dockOnStop`|Sends roomba home to dock when stopped|`true`|
```

I'm not sure what we put for the columns that don't have a default... just leave it blank?

**Review by karlvr — CHANGES_REQUESTED** [review]

**Comment by karlvr** [comment]
@rcoletti116 thank you, I hope you don't mind the constant change requests. I'm keen to make everything perfect!

File: src/accessory.ts (hunk 3 above)
Comment (rcoletti116) [reply src/accessory.ts:131]: Good question. Since it has a default value in the schema, anyone who installs it will get that value. It's currently not a required field in the scheme but we could make it so and that should cover most scenarios where it might be undefined by being left out of the config altogether.

Comment (karlvr) [reply src/accessory.ts:131]: @rcoletti116 I don't want to make it required... have you tested that having a default value in the schema means that the plugin receives that default value if it's missing from the schema? That was my question.

Comment (rcoletti116) [reply src/accessory.ts]: If it is missing from the config, the schema default does not pass through here. Added requested change to address this.

File: README.md (hunk 4 above)
Comment (rcoletti116) [reply README.md]: yes, I agree leave it blank if there is no default. additions made.

File: README.md [hunk 5]
```diff
@@ -67,24 +67,26 @@ configure your accessory using JSON:
...
+|---|-----------|---|
+|`accessory`|Loads this plugin. Must be set to `Roomba2`||
+|`name`|The name of your Roomba as it should appear in Homebridge and HomeKit||
+|`model`|The model of your Roomba as you'd like it to appear in HomeKit||
+|`serialnum`|The serial number as you'd like it to appear in HomeKit||
+|`blid`|The `blid` of your Roomba, obtained during setup||
+|`robotpwd`|The password for your Roomba, obtained during setup||
+|`ipaddress`|The [IP address](#troubleshooting) of your Roomba on your network||
+|`dockContactSensor`|Add a contact sensor to HomeKit that's _closed_ when Roomba is docked|`true`|
+|`runningContactSensor`|Add a contact sensor to HomeKit that's _open_ when Roomba is running||
```
Comment (karlvr) [inline README.md:86]: This must default to true or false, as do the other contact sensors, as it either appears or doesn't if it isn't set. I think it defaults false? Please check, as I'm going by memory while remote. And could you add defaults for those sensors to the config schema too?

File: config.schema.json [hunk 6]
```diff
@@ -37,7 +37,6 @@
             "dockContactSensor": {
                 "type": "boolean",
                 "title": "Show dock status as a contact sensor",
-                "default": true,
```
Comment (karlvr) [inline config.schema.json:40]: Why is this removed?

**Review by karlvr — CHANGES_REQUESTED** [review]

File: README.md [hunk 7]
```diff
@@ -98,11 +100,15 @@ as to provide HomeKit with Roomba's status only when it requests it.
 Click on any of the items below to expand the corresponding answer.
 
 <details>
-  <summary>I don't know what my Roomba's IP address is</summary>
```
Comment (karlvr) [inline README.md:101]: I'm a bit confused about these changes too. These are from a separate PR?

**Review by karlvr — CHANGES_REQUESTED** [review]

File: config.schema.json (hunk 6 above)
Comment (rcoletti116) [reply config.schema.json]: I had removed it to be consistent across all of the added sensors, when doing the docs. After looking back through the history I decided to put it back and must not have committed it.

File: README.md (hunk 7 above)
Comment (rcoletti116) [reply README.md:101]: I think they only show as changes now since you merged the other PR. I need to merge my branches. I'll fix it.

Comment (rcoletti116) [reply README.md:101]: Actually, I'm not sure why this is here...  the current version is correct.

**Comment by karlvr** [comment]
@rcoletti116 it's weird eh. I'll do a manual merge on it.

**Comment by karlvr** [comment]
@rcoletti116 OK, I rebased your PR, it ended up being a lot of messy commits so I squashed it up. I think the issue is that you're making the PR from a working branch. I reckon remove your branches, and always grab a fresh branch from this repository's main branch and build your PR from that, that will avoid the odd issues.

**Comment by karlvr** [comment]
Released in v1.3.0-beta.7
